### PR TITLE
eylultuncel-homework 2

### PR DESCRIPTION
**Pull request by eylultuncel**
new models added to the system

### CODE REVIEW

**Review by eibrahimarisoy — APPROVED** [review]

**Review by Gormezoglu — APPROVED** [review]

**Review by Tahakbas — APPROVED** [review]

File: ecommerce/customers/admin.py [hunk 1]
```diff
@@ -36,3 +36,31 @@ class CustomerAdmin(UserAdmin):
...
+
+
+@admin.register(Address)
+class ProductAdmin(admin.ModelAdmin):
+    search_fields = ("name", "phone")
+    list_display = ("name", "full_name", "line1", "line2", "phone", "district", "post_code", "city")
+
+
+@admin.register(City)
+class ProductAdmin(admin.ModelAdmin):
```
Comment (ttoygar) [inline ecommerce/customers/admin.py:56]: This class name should be CityAdmin by convention

File: ecommerce/customers/admin.py [hunk 2]
```diff
@@ -36,3 +36,31 @@ class CustomerAdmin(UserAdmin):
...
+    search_fields = ("name",)
+    list_display = ("name", "country",)
+    inlines = [AddressInline]
+
+
+@admin.register(Country)
+class ProductAdmin(admin.ModelAdmin):
```
Comment (ttoygar) [inline ecommerce/customers/admin.py:63]: This should be CountryAdmin.

File: ecommerce/customers/admin.py (hunk 1 above)
Comment (ttoygar) [inline ecommerce/customers/admin.py:56]: This class name should be CityAdmin by convention

File: ecommerce/customers/models.py [hunk 3]
```diff
@@ -67,3 +68,44 @@ def get_short_name(self):
...
+    def __str__(self):
+        return f"{self.name} - {self.country}"
+
+
+class Address(BaseAbstractModel):
+    name = models.CharField(max_length=255, verbose_name=_("Name"))
+    full_name = models.CharField(max_length=255, verbose_name=_("Full Name"))
+    line1 = models.CharField(max_length=255, verbose_name=_("Line1"))
+    line2 = models.CharField(max_length=255, verbose_name=_("Line2"))
+    phone = models.CharField(max_length=255, verbose_name=_("Phone"))
```
Comment (ttoygar) [inline ecommerce/customers/models.py:101]: It might be good to add some regex validators to this.

File: ecommerce/customers/models.py [hunk 4]
```diff
@@ -67,3 +68,44 @@ def get_short_name(self):
...
+    district = models.CharField(max_length=255, verbose_name=_("District"))
+    post_code = models.IntegerField(verbose_name=_("Post Code"))
```
Comment (ttoygar) [inline ecommerce/customers/models.py:103]: It might be good to add some regex validators to this too.

File: ecommerce/orders/models.py [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+
+    def __str__(self):
+        return f"{self.full_name} - {self.phone}"
+
+
+class ShippingAddress(BaseAbstractModel):
+    full_name = models.CharField(max_length=255, verbose_name=_("Full Name"))
+    line1 = models.CharField(max_length=255, verbose_name=_("Line1"))
+    line2 = models.CharField(max_length=255, verbose_name=_("Line2"))
+    phone = models.CharField(max_length=255, verbose_name=_("Phone"))
```
Comment (ttoygar) [inline ecommerce/orders/models.py:32]: It might be good to add some regex validators to this.

File: ecommerce/orders/models.py [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+    district = models.CharField(max_length=255, verbose_name=_("District"))
+    post_code = models.IntegerField(verbose_name=_("Post Code"))
```
Comment (ttoygar) [inline ecommerce/orders/models.py:34]: It might be good to add some regex validators to this.

File: ecommerce/payments/admin.py [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+
+from payments.models import Bank, BankAccount
+
+
+class BankAccountInline(admin.StackedInline):
+    model = BankAccount
+
+
+@admin.register(Bank)
+class ProductAdmin(admin.ModelAdmin):
```
Comment (ttoygar) [inline ecommerce/payments/admin.py:11]: Should be BankAdmin.

File: ecommerce/payments/admin.py [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+    search_fields = ("name",)
+    list_display = ("name",)
+    inlines = [BankAccountInline]
+
+
+@admin.register(BankAccount)
+class ProductAdmin(admin.ModelAdmin):
```
Comment (ttoygar) [inline ecommerce/payments/admin.py:18]: Should be BankAccountAdmin.

File: ecommerce/products/admin.py [hunk 9]
```diff
@@ -30,3 +30,9 @@ class StockAdmin(admin.ModelAdmin):
     list_display = ("product", "amount")
     search_fields = ("product__name", "product__sku")
     autocomplete_fields = ("product", )
+
+
+@admin.register(Category)
+class ProductAdmin(admin.ModelAdmin):
```
Comment (ttoygar) [inline ecommerce/products/admin.py:36]: Should be CategoryAdmin

**Review by ttoygar — APPROVED** [review]
It may be better to use slightly more docstrings.

File: ecommerce/baskets/enums.py [hunk 10]
```diff
@@ -0,0 +1,8 @@
+from django.db import models
+from django.utils.translation import gettext_lazy as _
+
+
+class Status(models.TextChoices):
```
Comment (upy) [inline ecommerce/baskets/enums.py:5]: BasketStatus dersek, olabilecek diger statuslarla karismasini engellemis oluruz.

File: ecommerce/customers/admin.py [hunk 11]
```diff
@@ -36,3 +36,31 @@ class CustomerAdmin(UserAdmin):
...
+    search_fields = ("name", "phone")
+    list_display = ("name", "full_name", "line1", "line2", "phone", "district", "post_code", "city")
+
+
+@admin.register(City)
+class ProductAdmin(admin.ModelAdmin):
+    search_fields = ("name",)
+    list_display = ("name", "country",)
+    inlines = [AddressInline]
```
Comment (upy) [inline ecommerce/customers/admin.py:59]: citynin adreslerini gormek adminde guzel bir fikir olmayabilir.

File: ecommerce/customers/models.py [hunk 12]
```diff
@@ -67,3 +68,44 @@ def get_short_name(self):
...
+    city = models.ForeignKey(City, verbose_name=_("City"), on_delete=models.PROTECT)
+
```
Comment (upy) [inline ecommerce/customers/models.py:105]: customer'i da FK olarak baglayabilir miyiz? odevde bahsetmeyi unutmusum.

File: ecommerce/orders/models.py [hunk 13]
```diff
@@ -0,0 +1,84 @@
...
+        verbose_name_plural = _("Order")
+
+    def __str__(self):
+        return f"{self.customer} - {self.basket}"
+
+
+class OrderBankAccount(BaseAbstractModel):
+    name = models.CharField(max_length=255, verbose_name=_("Name"))
+    iban = models.CharField(max_length=26, verbose_name=_("Iban"))
+    bank = models.ForeignKey(Bank, verbose_name="Bank", on_delete=models.PROTECT)
```
Comment (upy) [inline ecommerce/orders/models.py:63]: bank modelindeki bank name degistirilirse order zamanindaki secilen bank name tutarsiz olurmis olur.

File: ecommerce/products/models.py [hunk 14]
```diff
@@ -5,13 +5,25 @@
...
+
+
 class Product(BaseAbstractModel):
     sku = models.CharField(verbose_name=_("SKU"), max_length=100, unique=True)
     name = models.CharField(max_length=255, verbose_name=_("Name"))
     description = models.TextField(max_length=2000, verbose_name=_("Description"))
     color = models.CharField(
         choices=enums.Colors.choices, verbose_name=_("Color"), max_length=20)
     size = models.CharField(max_length=30, verbose_name=_("Size"))
+    categories = models.ManyToManyField(Category)
```
Comment (upy) [inline ecommerce/products/models.py:26]: verbose_name eklesek guzel olur.

**Review by upy — CHANGES_REQUESTED** [review]
yeni eklenen modellerin adminlerinde arama yaptirabilir miyiz? bazilarinda hata alacagiz gibi duruyor.

**Pull request closed without merging**